counterfit/core/logger.py

```diff
@@ -10,35 +23,26 @@ def __init__(self, **kwargs):
     def log(self, item):
         self.num_queries += 1
 
-class ListLogger(CFAttackLogger):
-    """Logs queries to a list.
+class JSONLogger(CFLogger):
+    """Logs queries to a json file saved to disk.
     """
     def __init__(self, **kwargs):
+        attack_id = kwargs["attack_id"]
+        ts = kwargs['ts']
+        self.num_queries = 0
+        self.filepath = pathlib.Path(f"attack_logs/{attack_id}_{ts}_logs.json")
         self.logs = []
-        self.queries = 0
+        self.filepath.parent.mkdir(parents=True, exist_ok=True)
 
     def log(self, item):
+        options = orjson.OPT_SERIALIZE_NUMPY | orjson.OPT_APPEND_NEWLINE
+        data = orjson.dumps(item,option=options)
+        with self.filepath.open('a+b') as fd:
```

Comment:
In this statement `self.filepath.open('a+b') ` do we need to pass ``a+b`` to the `open` function or is it a debug code?

Reply:
I believe this isn't a bug. `orjson.dumps(item,option=options)` returns type `bytes` so we need to append bytes to the buffer

Reply:
It's not related to the `orjson.dumps()`, I was referring to file open:
`with self.filepath.open('a+b') as fd:` In this line, we don't need to provide `a+b` string to the open function call. `self.filepath` is a `Path` object so we could apply method `open` on it without passing `a+b` string to it. That means, we could have `with self.filepath.open() as fd:` instead of with `self.filepath.open('a+b') as fd:`